### PR TITLE
Awesome books: with ES6

### PR DESCRIPTION
**Pull request by blueberry1312**
## Learning objectives
-Use ES6 modules to write modular JavaScript.
-Learn how to use proper ES6 syntax.
-Understand improvements that ES6 brings to JavaScript.
-Use npm as software packages system
## Changes
-Added modules js.
-Added linter configuration files.
-Updated readme file.

### CODE REVIEW

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# INVALID REVIEW REQUEST

Hi @blueberry1312 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that lacks a descriptive PR description.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+};
+
+const addRemoveButtons = (listBooks) => {
+  const buttons = document.getElementsByClassName('btn remove-btn');
+  for (let i = 0; i < buttons.length; i += 1) {
+    buttons[i].addEventListener('click', () => {
+      removeBookFromDOM(listBooks, i);
+    });
+  }
+};
```
Comment (Zuhaib042) [inline index.js:18]: - [ ] Good job implementing the remove book functionality, but the problem is that the book is not getting removed upon clicking until the user refreshes the page. kindly consider fixing it for a pleasant user experience.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+class Library {
+  constructor() {
+    this.books = [];
+  }
+
+  saveToLocalStorage() {
```
Comment (Zuhaib042) [inline index.js]: - [ ] Kindly consider using ES6 arrow functions in this class for all the methods instances as you have used ES6 arrow methods outside this class as it is project requirement to use ES6 arrow functions.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    localStorage.setItem('MY-Library', JSON.stringify(this.books));
+  }
+
+  getDataFromLocalStorage() {
```
Comment (Zuhaib042) [inline index.js]: - [ ] Kindly consider using ES6 arrow functions in this class for all the methods instances as you have used ES6 arrow methods outside this class as it is project requirement to use ES6 arrow functions.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      const data = JSON.parse(localStorage.getItem('MY-Library'));
+      if (data !== null) {
+        this.books = data;
+      }
+    } catch (error) {
+      this.saveToLocalStorage();
+    }
+  }
+
+  addBook() {
```
Comment (Zuhaib042) [inline index.js]: - [ ] Kindly consider using ES6 arrow functions in this class for all the methods instances as you have used ES6 arrow methods outside this class as it is project requirement to use ES6 arrow functions.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      this.getBooks();
+      form.reset();
+      document.getElementById('msg-add-book').innerHTML = 'New book added...';
+      setTimeout(() => {
+        document.getElementById('msg-add-book').innerHTML = '';
+      }, 3000);
+    }
+  }
+
+  getBooks() {
```
Comment (Zuhaib042) [inline index.js]: - [ ] Kindly consider using ES6 arrow functions in this class for all the methods instances as you have used ES6 arrow methods outside this class as it is project requirement to use ES6 arrow functions.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @blueberry1312 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
close

## Highlights
✅ Good ES6 usage
✅  Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi @blueberry1312   
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Well structured files :white_check_mark:
 * No Linter errors :white_check_mark:
 * Great ES6 usage :white_check_mark: 
 * Nice Design :white_check_mark: 
 * Good Job :+1: 
 
 Now you can merge this branch and move on :1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [x] N/A keep doing great job :clap: 

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**